lib/learning-object/learning-object.ts

```diff
@@ -142,195 +257,281 @@ export class LearningObject {
    *       reference itself is immutable, and elements can only be
    *       added and removed by the below functions
    */
-  goals: LearningGoal[];
+  get goals(): LearningGoal[] {
+    return this._goals;
+  }
+  /**
+   * Adds a new learning goal to this object.
+   * Returns the index of the new goal
+   */
+  addGoal(text: string): number {
+    const goal = new LearningGoal(text);
+    return this._goals.push(goal) - 1;
+  }
+  /**
+   * Removes the object's i-th learning goal.
+   * @param {number} index the index to remove from this object's goals
+   *
+   * @returns {LearningGoal} the goal which was removed
+   */
+  removeGoal(index: number): LearningGoal {
+    return this._goals.splice(index, 1)[0];
+  }
 
+  private _outcomes: LearningOutcome[];
   /**
    * @property {LearningOutcome[]} outcomes
    *       outcomes this object should enable students to achieve
    *
    */
-  outcomes: LearningOutcome[];
+  get outcomes(): LearningOutcome[] {
+    return this._outcomes;
+  }
+  /**
+   * Adds a passed outcome or new, blank learning outcome to this object.
+   * @returns {number} index of the outcome
+   */
+  addOutcome(outcome?: LearningOutcome): number {
+    const addingOutcome = outcome || new LearningOutcome();
```

Comment:
This should validate that the outcome provided is in fact an instance of a LearningOutcome before adding it.